### ISSUE TITLE
Deeplab, how calculate crop_size

### ISSUE THREAD

**Issue by essalahsouad** [tensorflow/models#6821]
### System information
- **What is the top-level directory of the model you are using**:
https://github.com/tensorflow/models/tree/master/research/deeplab
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:ubuntu 18.04
- **TensorFlow installed from (source or binary)**:binary
- **TensorFlow version (use command below)**:1.13.1 CPU
- **Bazel version (if compiling from source)**:N/A
- **CUDA/cuDNN version**:N/A
- **GPU model and memory**:N/A
- **Exact command to reproduce**:
`python deeplab / train.py \
  --logtostderr \
  --vis_split = "train" \
  --model_variant = "xception_65" \
  --atrous_rates = 6 \
  --atrous_rates = 12 \
  --atrous_rates = 18 \
  --output_stride = 16 \
  --decoder_output_stride = 4 \
  --train_crop_size = 513 \
  --train_crop_size = 513 \
  --train_batch_size = 64 \
  --fine_tune_batch_norm = false \
  --tf_initial_checkpoint = "./ data / init_models / deelabv3_pascal_train_aug \ model.ckpt"
  --initialize_last_layer = false \
  --last_layers_contain_logits_only = true \
  --train_logdir = "./ data / log / train" \
  --dataset_dir = "./ data / tfrecord" \
  --dataset = "project"`
`python deeplab / eval.py \
  --logtostderr \
  --eval_split = "val" \
  --model_variant = "xception_65" \
  --atrous_rates = 6 \
  --atrous_rates = 12 \
  --atrous_rates = 18 \
  --output_stride = 16 \
  --decoder_output_stride = 4 \
  --eval_crop_size = 513 \
  --eval_crop_size = 513 \
  --max_resize_value = 512 \
  --min_resize_value = 128 \
  --checkpoint_dir = "./ data / log / train" \
  --eval_logdir = "./ data / log / eval" \
  --dataset_dir = "./ data / tfrecord" \
  --max_number_of_evaluations = 1 \
  --eval_interval_secs = 0 \
  --dataset = "project"`
`python deeplab / vis.py \
  --logtostderr \
  --vis_split = "val" \
  --model_variant = "xception_65" \
  --atrous_rates = 6 \
  --atrous_rates = 12 \
  --atrous_rates = 18 \
  --output_stride = 16 \
  --decoder_output_stride = 4 \
  --vis_crop_size = 513 \
  --vis_crop_size = 513 \
  --max_resize_value = 512 \
  --min_resize_value = 128 \
  --checkpoint_dir = "./ data / log / train" \
  --vis_logdir = "./ data / log / vis" \
  --dataset_dir = "./ data / tfrecord" \
  --max_number_of_iterations = 1 \
  --eval_interval_secs = 0 \
  --dataset = "project"`
### Describe the problem
I tried to train deeplab model on my custom dataset, but I don't know how to calculate the crop_size for training, verification ,visaltion , and test

train_crop_size
eval_crop_size
vis_crop_size
crop_size
For a 448 * 300 image and how they were calculated?

**Comment by aquariusjay** [comment]
Hi essalahsouad,

Thanks for bringing up this issue.
Please check Q8 in [FAQ](https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/faq.md) for reference.
In your case, you could set train_crop_size=eval_crop_size=vis_crop_size=449*305.

Cheers,

**Comment by essalahsouad** [comment]
@aquariusjay  
 thank you for your very full reply. 
but sir I would ask you for the rule to calculate the crop_size because I don't really understand how is it going and the values that I put in the issues are just an example. my real data size is 400*300, I would appreciate your help.
and I want to make sure if the crop)size for the RGB image or for the label image

**Comment by Qhkm92** [comment]
Yeah i'm also confused with that. Answers is much appreciated!
